src/components/find-on-shelf/FindOnShelfModal.tsx

```diff
@@ -1,52 +1,87 @@
 import * as React from "react";
 import { FC } from "react";
-import { ManifestationsSimpleFieldsFragment } from "../../core/dbc-gateway/generated/graphql";
+import { isAnyManifestationAvailableOnBranch } from "../../apps/material/helper";
 import { useGetHoldingsV3 } from "../../core/fbs/fbs";
 import {
   convertPostIdToFaustId,
   creatorsToString,
   filterCreators,
-  flattenCreators
+  flattenCreators,
+  getManifestationsPids
 } from "../../core/utils/helpers/general";
 import Modal from "../../core/utils/modal";
 import { useText } from "../../core/utils/text";
+import { Manifestation, Work } from "../../core/utils/types/entities";
+import { ManifestationHoldings } from "../../core/utils/types/find-on-shelf";
 import { FaustId, Pid } from "../../core/utils/types/ids";
 import Disclosure from "../material/disclosures/disclosure";
+import FindOnShelfManifestationList from "./FindOnShelfManifestationList";
 
 export const findOnShelfModalId = (faustId: FaustId) =>
   `find-on-shelf-modal-${faustId}`;
 
 export interface FindOnShelfModalProps {
-  manifestation: ManifestationsSimpleFieldsFragment;
+  manifestations: Manifestation[];
+  workTitles: string[];
+  authors: Work["creators"];
+  pid?: Pid;
 }
 
-const FindOnShelfModal: FC<FindOnShelfModalProps> = ({ manifestation }) => {
+const FindOnShelfModal: FC<FindOnShelfModalProps> = ({
+  manifestations,
+  workTitles,
+  authors,
+  pid
+}) => {
   const t = useText();
-  const {
-    pid,
-    creators,
-    titles: { main: mainTitle }
-  } = manifestation;
-  const faustId = convertPostIdToFaustId(pid as Pid);
-  const author =
-    creatorsToString(
-      flattenCreators(filterCreators(creators, ["Person"])),
-      t
-    ) || t("creatorsAreMissingText");
+  const pidArray = getManifestationsPids(manifestations);
+  const faustIdArray = pidArray.map((manifestationPid) =>
+    convertPostIdToFaustId(manifestationPid as Pid)
+  );
   const { data, isError, isLoading } = useGetHoldingsV3({
-    recordid: [faustId]
+    recordid: faustIdArray
   });
+  const author =
+    creatorsToString(flattenCreators(filterCreators(authors, ["Person"])), t) ||
+    t("creatorsAreMissingText");
+  const title = workTitles.join(", ");
+
+  const modalFaustId = pid ? convertPostIdToFaustId(pid as Pid) : undefined;
```

Comment:
Shouldn't be necessary with `as Pid`

Reply:
Good stuff! :) Loving the new Manifestation type :)